api/src/test/java/org/apache/iceberg/expressions/TestStrictMetricsEvaluator.java

```diff
@@ -627,4 +638,47 @@ public void testIntegerNotIn() {
     shouldRead = new StrictMetricsEvaluator(SCHEMA, notIn("no_nulls", "abc", "def")).eval(FILE);
     assertThat(shouldRead).as("Should not match: no_nulls field does not have bounds").isFalse();
   }
+
+  @Test
+  public void testEvaluateOnNestedColumns() {
+    boolean shouldRead =
+        new StrictMetricsEvaluator(
+                SCHEMA, greaterThanOrEqual("struct.nested_col_no_stats", INT_MIN_VALUE))
+            .eval(FILE);
+    assertThat(shouldRead).as("Should not match: nested column").isFalse();
+
+    shouldRead =
+        new StrictMetricsEvaluator(
+                SCHEMA, lessThanOrEqual("struct.nested_col_no_stats", INT_MAX_VALUE))
+            .eval(FILE);
+    assertThat(shouldRead).as("Should not match: nested column").isFalse();
+
+    shouldRead =
+        new StrictMetricsEvaluator(SCHEMA, isNull("struct.nested_col_no_stats")).eval(FILE);
+    assertThat(shouldRead).as("Should not match: nested column").isFalse();
+
+    shouldRead =
```

Comment:
Is it possible to separate these into 2 test methods, one for no stats and the other with stats? Could we also update the messages to be more precise, instead of just "Should not match: nested column", I think "isNull nested column should not match"